File: OOPs/Abstract & Concrete Methods/abstract_concrete_method.cpp
#include<bits\stdc++.h>
using namespace std;

class Shape {
    public:
        // Abstract method
        virtual void draw() = 0;

        // Concrete method
        void description() {
            cout << "This is a shape" << endl;
        }
};

class Circle : public Shape {
    public:
        void draw() override {
            cout << "Drawing a Circle" << endl;
        }
};

int main() {
    Shape* shape = new Circle();
    shape->description(); // Concrete method
    shape->draw();        // Abstract method implementation

    delete shape;
    return 0;
}

File: OOPs/Abstract & Concrete Methods/abstract_concrete_method.py
from abc import abstractmethod

class Shape:
    # Abstract Method
    @abstractmethod
    def draw(self):
        pass
    
    # Concrete Method
    def paint(self):
        print('Painting the shape')

class Circle(Shape):
    def draw(self):
        print('Drawing a circle')

if __name__ == '__main__':
    c = Circle()
    c.draw()        # Abstract method implementation
    c.paint()       # Concrete method